### PR TITLE
add release config

### PR DESCRIPTION
**Pull request by philippdormann**
- automated changelog containing commits + diff from last release
- automated version bumping
- tag creation
    - might be used to trigger GitHub workflows/ GitHub Actions for Docker releases in the future?
- semver support

### CODE REVIEW

**Comment by Saibamen** [comment]
@philippdormann: Please resolve conflicts

@louislam: +1 for this. This will take care of correct releases in GitHub.
Current release is wrong:
image

**Comment by philippdormann** [comment]
thanks for stepping in and "+1'ing" 😉

sure thing :) will resolve the conflicts shortly

File: package.json [hunk 1]
```diff
@@ -40,6 +41,26 @@
         "@vue/compiler-sfc": "^3.0.5",
         "core-js": "^3.15.2",
         "sass": "^1.35.1",
-        "vite": "^2.3.7"
+        "vite": "^2.3.7",
+        "auto-changelog": "2.3.0",
+        "release-it": "14.10.0"
```
Comment (Saibamen) [inline package.json:46]: Please move it to devDependencies. We don't want to have unused packaged in PROD Docker image (https://github.com/louislam/uptime-kuma/pull/122)

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (nicolaiort) [reply package.json:46]: The two dependencies you mentioned are already devDependencies :)

**Review by Saibamen — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (Saibamen) [reply package.json]: Sorry, I didn't see full file before :P

**Comment by louislam** [comment]
Since my release approach is a bit special, I have my own script for that. Will be back to this later.

**Comment by Saibamen** [comment]
@philippdormann: Can you check https://github.com/louislam/uptime-kuma/blob/master/extra/update-version.js and https://github.com/louislam/uptime-kuma/blob/master/extra/mark-as-nightly.js to see how current release looks like?

**Comment by philippdormann** [comment]
well of course I took a look at it and
tbh don't like it at all - that's why this PR was created 😄😬

**Comment by Saibamen** [comment]
.

**Comment by louislam** [comment]
I prefer to write changelog manually and currently `npm run release-final` did a great job for me. 

So it will be closed, thank you for the pr.

**Pull request closed without merging**